### DISCUSSION TITLE
Inputting an enum

### DISCUSSION THREAD

**Discussion by twiclo** [graphql-rust/juniper#1112] [Q&A] [answered]
I have the following setup:
```
struct NewLocation {
  address: Option<NewAddress>,
  coords: Option<NewCoordinate>
}
```

I want to ensure that you can't submit a `NewLocation` unless you provide either a `NewAddress` or a `NewCoordinate`. I thought that an Enum would be great for that but I'm not seeing a way to implement an Enum as an input object in The Book. I do however see an Enum here: https://docs.rs/juniper/latest/juniper/enum.InputValue.html

Some guidance would be much appreciated

**Comment by ilslv** [comment, marked as answer]
GraphQL enums are like C-style enums without inner type, so what you are looking for are [GraphQL unions][1]. Unfortunately as of now GraphQL [doesn't support unions as input types][2]. So for now you have to validate your input objects manually. 




[1]: https://graphql-rust.github.io/juniper/master/types/unions.html
[2]: https://github.com/graphql/graphql-spec/pull/825
